File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/TypeFactory.java
package ch.rmy.android.http_shortcuts.variables.types;

import ch.rmy.android.http_shortcuts.realm.models.Variable;

public class TypeFactory {

    public static BaseVariableType getType(String type) {
        switch (type) {
            case Variable.TYPE_TEXT:
                return new TextType();
            case Variable.TYPE_NUMBER:
                return new NumberType();
            case Variable.TYPE_PASSWORD:
                return new PasswordType();
            case Variable.TYPE_TOGGLE:
                return new ToggleType();
            case Variable.TYPE_SELECT:
                return new SelectType();
            case Variable.TYPE_COLOR:
                return new ColorType();
            case Variable.TYPE_CONSTANT:
            default:
                return new ConstantType();
        }
    }

}

File: HTTPShortcuts/app/src/main/java/ch/rmy/android/http_shortcuts/variables/types/TypeFactory.kt
package ch.rmy.android.http_shortcuts.variables.types

import ch.rmy.android.http_shortcuts.realm.models.Variable

object TypeFactory {

    fun getType(type: String): BaseVariableType {
        return when (type) {
            Variable.TYPE_TEXT -> TextType()
            Variable.TYPE_NUMBER -> NumberType()
            Variable.TYPE_PASSWORD -> PasswordType()
            Variable.TYPE_TOGGLE -> ToggleType()
            Variable.TYPE_SELECT -> SelectType()
            Variable.TYPE_COLOR -> ColorType()
            Variable.TYPE_DATE -> DateType()
            else -> ConstantType()
        }
    }

}
